sopel/cli/run.py

```diff
@@ -544,159 +466,33 @@ def command_restart(opts):
-    old_pid = get_running_pid(pid_file_path)
-
-    if old_pid is not None and tools.check_pid(old_pid):
-        if not opts.quit and not opts.kill and not opts.restart:
-            tools.stderr(
-                'There\'s already a Sopel instance running with this config file')
-            tools.stderr(
-                'Try using either the `sopel stop` command or the `sopel restart` command')
-            return ERR_CODE
-        elif opts.kill:
-            tools.stderr(
-                'WARNING: option -k/--kill is deprecated; '
-                'use `sopel stop --kill` instead')
-            tools.stderr('Killing the Sopel')
-            os.kill(old_pid, signal.SIGKILL)
-            return
-        elif opts.quit:
-            tools.stderr(
-                'WARNING: options -q/--quit is deprecated; '
-                'use `sopel stop` instead')
-            tools.stderr('Signaling Sopel to stop gracefully')
-            if hasattr(signal, 'SIGUSR1'):
-                os.kill(old_pid, signal.SIGUSR1)
-            else:
-                # Windows will not generate SIGTERM itself
-                # https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/signal
-                os.kill(old_pid, signal.SIGTERM)
-            return
-        elif opts.restart:
-            tools.stderr(
-                'WARNING: options --restart is deprecated; '
-                'use `sopel restart` instead')
-            tools.stderr('Asking Sopel to restart')
-            if hasattr(signal, 'SIGUSR2'):
-                os.kill(old_pid, signal.SIGUSR2)
-            else:
-                # Windows will not generate SIGILL itself
-                # https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/signal
-                os.kill(old_pid, signal.SIGILL)
-            return
-    elif opts.kill or opts.quit or opts.restart:
-        tools.stderr('Sopel is not running!')
-        return ERR_CODE
-
-    if opts.daemonize:
-        child_pid = os.fork()
-        if child_pid != 0:
-            return
-    with open(pid_file_path, 'w') as pid_file:
-        pid_file.write(str(os.getpid()))
-
-    # Step Four: Initialize and run Sopel
-    ret = run(settings, pid_file_path)
-    os.unlink(pid_file_path)
-    if ret == -1:
-        os.execv(sys.executable, ['python'] + sys.argv)
-    else:
-        return ret
-
-
-def main(argv=None):
-    """Sopel run script entry point.
-
-    :param list argv: command line arguments
-    """
-    try:
-        # Step One: Parse The Command Line
-        parser = build_parser()
-
-        # make sure to have an action first (`legacy` by default)
-        # TODO: `start` should be the default in Sopel 8
-        argv = argv or sys.argv[1:]
-        if not argv:
-            argv = ['legacy']
-        elif argv[0].startswith('-') and argv[0] not in ['-h', '--help']:
-            argv = ['legacy'] + argv
-
-        opts = parser.parse_args(argv)
-
         # Step Two: "Do not run as root" checks
```

Comment:
Step Two, the root check, doesn't need to be inside the outer `try`, does it? It's a perfect opportunity to flatten the nesting structure that, AFAICT, exists for no good reason. Moving this check to happen before handling of `--version` would match the current (7.1.x) behavior.

For previously-discussed (many times) GitHub reasons, I can't select and attach this note to the whole block. Two lines in the middle that are hidden in the diff view prevent doing so.